src/Graphics/Glyphs/GlyphGeom.cc

```diff
@@ -311,29 +316,29 @@ void GlyphGeom::generateCylinder(const Point& p1, const Point& p2, double radius
   Vector p;
   for (int strips = 0; strips <= num_strips; strips++)
   {
-    size_t offset = static_cast<size_t>(numVBOElements_);
+    size_t offset = static_cast<size_t>(geomData_.numVBOElements);
     p = std::cos(2. * M_PI * strips / num_strips) * u +
       std::sin(2. * M_PI * strips / num_strips) * crx;
     p.normalize();
     Vector normals(((p2-p1).length() * p + (r2-r1)*n).normal());
 
-    points_.push_back(r1 * p + Vector(p1));
-    colors_.push_back(color1);
-    normals_.push_back(normals);
-    numVBOElements_++;
-    points_.push_back(r2 * p + Vector(p2));
-    colors_.push_back(color2);
-    normals_.push_back(normals);
-    numVBOElements_++;
-
-    indices_.push_back(0 + offset);
-    indices_.push_back(1 + offset);
-    indices_.push_back(2 + offset);
-    indices_.push_back(2 + offset);
-    indices_.push_back(1 + offset);
-    indices_.push_back(3 + offset);
+    geomData_.points.push_back(r1 * p + Vector(p1));
+    geomData_.colors.push_back(color1);
+    geomData_.normals.push_back(normals);
+    geomData_.numVBOElements++;
+    geomData_.points.push_back(r2 * p + Vector(p2));
+    geomData_.colors.push_back(color2);
+    geomData_.normals.push_back(normals);
+    geomData_.numVBOElements++;
+
+    geomData_.indices.push_back(0 + offset);
```

Comment:
Now that there's a new class, this block (334-339) can be a function of `GlyphConstructionParameters`. Will again reduce duplication and you can introduce a name to explain the magic numbers.

Reply:
There's now a function addVertex(point, normal, color) which replaces adding each of those to a vector and also increments numVBOElements